EventSourcing.Core/Aggregates/AggregateTransaction.cs

```diff
@@ -0,0 +1,75 @@
+using Microsoft.Extensions.Logging;
+
+namespace EventSourcing.Core;
+
+public class AggregateTransaction<TBaseEvent> : IAggregateTransaction<TBaseEvent> where TBaseEvent : Event, new()
+{
+  private readonly IEventTransaction<TBaseEvent> _eventTransaction;
+  private readonly ISnapshotStore<TBaseEvent> _snapshotStore;
+  private readonly ILogger<AggregateService<TBaseEvent>> _logger;
+  private readonly List<Aggregate<TBaseEvent>> _aggregates = new();
+
+  public AggregateTransaction(
+    IEventTransaction<TBaseEvent> eventTransaction,
+    ISnapshotStore<TBaseEvent> snapshotStore,
+    ILogger<AggregateService<TBaseEvent>> logger)
+  {
+    _eventTransaction = eventTransaction;
+    _snapshotStore = snapshotStore;
+    _logger = logger;
+  }
+  
+  public async Task<TAggregate> PersistAsync<TAggregate>(TAggregate aggregate, CancellationToken cancellationToken = default)
```

Comment:
Maybe AddAsync is a better name?

Reply:
Maybe it is, I'm uncertain about it, one could make argument for both the AddAsync and PersistAsync cases

Reply:
At least 'Add' is consistent with a few lines below: await _eventTransaction.AddAsync(aggregate.UncommittedEvents.ToList(), cancellationToken);